Model/Util/DebugMode.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Razorpay\Magento\Model\Util;
+
+class DebugMode
+{
+    /**
+     * @var \Razorpay\Magento\Model\Config
+     */
+    protected $config;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected $logger;
+
+    protected $isDebugModeEnabled;
+
+    public function __construct(
+        \Razorpay\Magento\Model\Config $config,
+        \Psr\Log\LoggerInterface $logger
+    )
+    {
+        $this->config                   = $config;
+        $this->logger                   = $logger;
+
+        $this->isDebugModeEnabled = $this->config->isDebugModeEnabled();
+    }
+
+    public function log($message)
+    {
+        if($this->isDebugModeEnabled)
+        {
+            $this->logger->info($message);
+        }
+    }
+}
```

Comment:
We are working in the *nix environment, make sure to leave empty line at the EOF (PSR)